osu.Game/Screens/Play/SubmittingPlayer.cs

```diff
@@ -150,7 +150,7 @@ private Task submitScore(Score score)
                 return scoreSubmissionSource.Task;
 
             // if the user never hit anything, this score should not be counted in any way.
-            if (!score.ScoreInfo.Statistics.Any(s => s.Key.IsHit() && s.Value > 0))
+            if (!score.ScoreInfo.Statistics.Any(s => s.Key.IsHit() && !s.Key.IsBonus() && s.Value > 0))
```

Comment:
Could this be rewritten to use `IsBasic`? Or is that not valid here due to the `IsTick` part?

https://github.com/ppy/osu/blob/8078a8c1f8b21709d3741e9dcb83636fecb295a9/osu.Game/Rulesets/Scoring/HitResult.cs#L167-L168

Finding this line a bit hard to read with the outer `!` and now an inner `!`.

Reply:
I was initially going to use `IsBasic` but realized osu-web allowing tick judgements, maybe fine to just not submit on tick-only as well?

Reply:
In all existing usages, I think it makes sense to not submit in such cases. Ticks are usually followed by a final `Basic` type.

Reply:
It would have to be written as `s.Key.IsHit() && (s.Key.IsBasic() || s.Key.IsTick())` since "tick" judgements aren't "basic"

Reply:
Do you think we need only-ticks-submitting? I can't think of a scenario we'd want that. Generally the use case of submission is incrementing hit stats, but ticks don't count to those.

Reply:
In the context of how the game currently works, sure, that'd be fine. As with bonus, I'm not sure how correct it is though.

For example I could easily envision a ruleset where you get bonus points for collecting things, but there's no actual hitobjects with normal judgements.

The decisions in this PR are completely arbitrary, so long as you understand that.

Reply:
Yep, I understand that. I think it's better to play conservatively and deal with it when it comes up, if ever.